client/components/mma/cancel/CancellationJourneyFunnel.tsx

```diff
@@ -58,7 +58,10 @@ export const CancellationJourneyFunnel = () => {
 		return <Navigate to="./landing" state={{ ...routerState }} />;
 	}
 
-	if (!productType.cancellation?.reasons) {
+	if (
+		Object.hasOwn(productType, 'cancellation') &&
```

Comment:
This is the line of code that fixes the main bug

Reply:
the property is on the object, not it's prototype.

Previously we were checking against 2 scenarios:
- with cancellation reasons
- without cancellation reasons

What we need to do is check for 3 scenarios:
- with cancellation reasons
- without cancellation reasons
- without a cancellation object at all